### ISSUE TITLE
Create a docker image for testing that has a reasonable environment

### ISSUE THREAD

**Issue by gonuke** [svalinn/parastell#57]
To better support #12 we should confirm the best way to define a Dockerfile that makes a robust parastell testing environment.  The first version can skip the Cubit component, which makes it only really useful for source mesh testing.  We can then use that Dockerfile to build a CI image for testing in #13

**Comment by Edgar-21** [comment]
Would it be a good idea to use an openmc image as the base, or start with something like a ubuntu 22.04?

**Comment by gonuke** [comment]
You guys know best.  Are there conflicts with the openmc dependencies?  We don't strictly need OpenMC in the image for testing....

**Comment by connoramoreno** [comment]
@Edgar-21, I'd say we should begin with a Dockerfile that uses Ubuntu 22.04 but only add the dependencies specific to ParaStell

**Comment by connoramoreno** [comment]
The trick will be getting Cubit integrated into the Dockerfile but I believe @gonuke is working on a solution to that with Coreform

**Comment by Edgar-21** [comment]
I can take a crack at this unless someone else has started already?

**Comment by gonuke** [comment]
I just sent a license file in Slack with some pointers

**Comment by gonuke** [comment]
Also, here's my first pass at a working Dockerfile:
```
FROM ubuntu:22.04 as base
ARG license_server=144.92.12.212

ENV TZ=America/Chicago
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

RUN apt update --fix-missing

RUN apt install -y wget libglu1-mesa libgl1-mesa-glx fontconfig-config fonts-dejavu-core libbrotli1 libbsd0 libexpat1 libfontconfig1 libfreetype6 libice6 libjpeg-turbo8 libjpeg8 libmd0 libmotif-common libpng16-16 libsm6 libx11-6 libx11-data libxau6 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-render0 libxcb-shm0 libxcb-util1 libxcb-xfixes0 libxcb-xinerama0 libxcb-xkb1 libxcb1 libxdmcp6 libxext6 libxft2 libxkbcommon-x11-0 libxkbcommon0 libxm4 libxmu6 libxrender1 libxt6 ucf x11-common xkb-data

# Use one of the following lines.  
# 1. The first makes sense in general and downloads the DEB file from Coreform.
# 2. The second makes sense if you already have it downloaded and want to avoid that network traffic (e.g. testing)
#RUN wget -O /cubit.deb https://f002.backblazeb2.com/file/cubit-downloads/Coreform-Cubit/Releases/Linux/Coreform-Cubit-2024.3%2B46968-Lin64.deb
COPY ./cubit.deb /cubit.deb

# install Cubit
RUN dpkg -i /cubit.deb

# Create a license file from the build argument
RUN echo "HOST ${license_server} ANY 5053\nISV csimsoft\n" > /opt/Coreform-Cubit-2024.3/bin/licenses/remote.lic

# run Cubit once to answer the interactive question
RUN echo y | /opt/Coreform-Cubit-2024.3/bin/coreform_cubit -nographics

# delete the license file
RUN rm /opt/Coreform-Cubit-2024.3/bin/licenses/remote.lic

# To use this docker image
# 1. run image
# 2. write license file as above
# 3. use cubit
# 4. when docker image ends, license file will disapper
#
# If running from Github Action
# 1. run image
# 2. set environment variable from GITHUB secret with license server IP
# 3. write license file as above
# 4. run tests
# 5. when docker image ends, license file will disappear
```

**Comment by gonuke** [comment]
This should probably include the `environment.yml` file as necessary

**Comment by gonuke** [comment]
Closed by #58
